### PR TITLE
IGNITE-17809 Improve passing arguments to docker container

### PR DESCRIPTION
**Pull request by valepakh**
https://issues.apache.org/jira/browse/IGNITE-17809

### CODE REVIEW

File: modules/runner/src/main/java/org/apache/ignite/app/IgniteCliRunner.java [hunk 1]
```diff
@@ -17,38 +17,149 @@
...
+import picocli.CommandLine.Option;
+import picocli.CommandLine.TypeConversionException;
 
 /**
  * The main entry point for run new Ignite node from CLI toolchain.
  */
-public class IgniteCliRunner {
+@SuppressWarnings("FieldMayBeFinal")
+@Command(name = "runner")
+public class IgniteCliRunner implements Callable<CompletableFuture<Ignite>> {
```
Comment (PakhomovAlexander) [inline modules/runner/src/main/java/org/apache/ignite/app/IgniteCliRunner.java:48]: I would like to ask you to add the java doc that explains how exactly parameters are overridden each and what rules are applied. For example, `--node-name` will override the env variable `IGNITE_NODE_NAME`.

File: modules/runner/src/main/java/org/apache/ignite/app/IgniteCliRunner.java [hunk 2]
```diff
@@ -17,38 +17,150 @@
...
+    private static final String CONFIG_PATH = "--config-path";
+
+    private static final String CONFIG_STRING = "--config-string";
```
[the remark this answers is not in the document]
Comment (valepakh) [reply modules/runner/src/main/java/org/apache/ignite/app/IgniteCliRunner.java:51]: Ah, it's a leftover from another workaround.

File: packaging/scripts/bootstrap-config [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
-WORK_PATH=$IGNITE_HOME/work
 LOGS_PATH=$IGNITE_HOME/log
 LIBS_PATH=$IGNITE_HOME/lib
 BIN_PATH=$IGNITE_HOME/bin
 CONFIG_PATH=$IGNITE_HOME/etc
 
-IGNITE_CONFIG_FILE=$CONFIG_PATH/ignite-config.conf
 LOG_OUT_FILE=$LOGS_PATH/${NODE_NAME}.log
+
+# Export these variables so that IgniteCliRunner can use them as default values
```
[the remark this answers is not in the document]
Comment (valepakh) [reply packaging/scripts/bootstrap-config:8]: I thought that this could also be benefitial for the zip startup script.

**Comment by PakhomovAlexander** [comment]
LGTM;

**Pull request closed without merging**